dart/dynamics/Linkage.hpp

```diff
@@ -127,6 +127,27 @@ class Linkage : public ReferentialSkeleton
     /// entry in mTargets) will be halted if it reaches any entry in mTerminal
     std::vector<Terminal> mTerminals;
 
+    /// Constructs an empty criteria that will lead to creating an empty Linkage
+    Criteria() = default;
+
+    /// Utility constructor to create a contiguous sequence of BodyNodes that
+    /// doesn't include branches in it.
+    ///
+    /// The start and target can be interchangeably set if both are included in
+    /// the sequence.
+    ///
+    /// \param[in] start The first BodyNode in the sequence. If \c nullptr is
+    /// passed the sequence will expand from \c target to the root.
+    /// \param[in] target The second BodyNode in the sequence. If \c nullptr is
+    /// passed the sequence will expand from \c start to the root.
+    /// \param[in] includeBoth Set this to true if both \c start and \c target
+    /// should be included. Otherwise, whichever is upstream of the other will
+    /// be left out of the sequence.
+    Criteria(
+        BodyNode* start,
+        BodyNode* target,
+        bool includeBoth = false);
```

Comment:
I realize that this is just copying from `Chain.hpp` and that I'm probably the one that came up with the `includeBoth` concept, but I'm not actually sure that I understand the motive. Why should the options be:

1. Include both the start and the target, or
2. Leave out whichever one is upstream

I don't really see what the significance of the upstream link would be that we would want to leave it out. The only thing I can think of is maybe we want to "start" from a body node that's at a fork and then "target" a body node that's at the end of a branch. I suppose that would be helpful for constructing chains within multi-arm skeletons.

I'm fine with leaving it as it is and merging it, but I thought I'd go ahead and voice my own confusion at an API that I probably made myself. If anyone else finds it confusing, we should probably consider tweaking it somehow.

Reply:
> I realize that this is just copying from `Chain.hpp`

For the record, it is identical except that `mChain` option is false in this constructor. That's why I didn't make Chain use this constructor (maybe I should let Chain just use this constructor and change the option to true?).

Regarding `includeBoth`, I assumed that this option is about whether it includes the parent joint of the upstream body or not because the joint is technically not in between `start` and `target` bodies. In this case, we could change the option name and the document more clear.

Reply:
Oh good call! You're exactly right: it's about whether to include the parent joints of both the start and the target. A more descriptive name would definitely be helpful. Maybe `includeUpstreamParentJoint`?